### DISCUSSION TITLE
DATABASE_PASS stored in clear text in docker-compose. Support of docker secret?

### DISCUSSION THREAD

**Discussion by CypZ** [teslamate-org/teslamate#2863] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

In my opinion the database password should be stored as a docker secret, esp. now that docker supports it outside of a swarm.

For people like me who store and version their compose files on GitHub, it just doesn't look right to see passwords as clear text in there, even on private repos.

See #58 for similar issue on the API token

version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY= #insert a secure key to encrypt your Tesla API tokens
      - DATABASE_USER=teslamate
      - DATABASE_PASS= #insert your secure database password!

### Expected Behavior

version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY= #insert a secure key to encrypt your Tesla API tokens
      - DATABASE_USER=teslamate
      - DATABASE_PASS_FILE=/run/secrets/teslamate_db_password
    secrets:
      - teslamate_db_password

[...]

secrets:
  teslamate_db_password:
    file: /home/pi/my_secrets/teslamate_db_password

### Relevant log output

```shell
N/A
```

### Type of installation

Docker

### Version

v1.27.1

**Comment by adriankumpf** [comment]
This is a good idea. What is the minimum Docker version required for Docker secrets? 

It would have to be supported by older Docker versions as well, so that we can recommend it.

**Comment by CypZ** [comment]
According to [github.com/docker/compose/pull/4368](https://github.com/docker/compose/pull/4368) the min version for compose file is 3.1 and API is 1.13.0.

**Reply by sderungs** [reply]
this would actually be a great addition to security for both variables DATABASE_PASS and ENCRYPTION_KEY if they had a corresponding *_FILE counterpart to use with Docker secrets

**Comment by CypZ** [comment]
yea in the meantime I just moved all environments variables out of the compose file and centralized (and factored) them in the default .env file (which can be gitignored) as suggested in #58

![image](https://user-images.githubusercontent.com/18141587/208878283-a91e1093-74ee-43d3-803a-19d89e550170.png)

Not as secure as Secrets but still better than keeping values in clear text in the compose file.

**Reply by sderungs** [reply]
unfortunately, the .env file is not supported in a Docker Swarm setup (to my knowledge) :-(
